c2ctask/src/containerMapping.ts

```diff
@@ -0,0 +1,86 @@
+import tl = require('azure-pipelines-task-lib/task');
+import * as common from './common';
+import { IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+import { ICodeToCloud } from './c2c';
+
+
+/*
+* Class for Container Mapping functionality in Code to Cloud Decorator task.
+*/
+export class ContainerMapping implements ICodeToCloud {
+    private readonly taskType: common.TaskType;
+    private readonly version: string;
+    private readonly sectionDelim: string = ":::";
+    private readonly preJobStartTime: string = "PREJOBSTARTTIME";
+    private readonly imageOptions: IExecOptions = {
+        silent: true
+    };
+
+    constructor(inputString: string | undefined) {
+        this.taskType = inputString as common.TaskType;
+        this.version = common.getTaskVersion();
+    }
+
+    /*
+    * Set the start time of the job run.
+    */
+    private runPre() {
+        const startTime = new Date().toISOString();
+        tl.setVariable(this.preJobStartTime, startTime);
+    }
+
+    /*
+    * Using the start time, fetch the docker events and docker images in this job run and log the encoded output
+    */
+    private runPost() {
+        let startTime = tl.getVariable(this.preJobStartTime);
+        if (startTime == undefined) {
+            throw new Error(this.preJobStartTime + " variable not set");
+        }
+        let data : string[] = [];
+        data.push(common.execTaskCmdSync("docker", ["--version"], this.imageOptions));
```

Comment:
Why not check if docker is installed first? I see that you're capturing the error in execTaskCmdSync, but it would be better to differentiate between the two, since it is not really an error condition for the customer to not be using docker.

Reply:
If we dont have `docker`, `docker --version` would fail and stop execution. I think this is a good enough way to determine if docker is installed or not.

> since it is not really an error condition for the customer to not be using docker

None of these errors would be reported back to the customer, all they're going to see is success. We will parse this on our backend and capture it there.